prov/gni/src/gnix_cq.c

```diff
@@ -612,8 +623,46 @@ DIRECT_FN STATIC ssize_t gnix_cq_readerr(struct fid_cq *cq,
 	}
 
 	event = container_of(entry, struct gnix_cq_entry, item);
-
-	memcpy(buf, event->the_entry, sizeof(struct fi_cq_err_entry));
+	gnix_cq_err = event->the_entry;
+#if 0
+	fabric = container_of(&cq_priv->domain->fabric->fab_fid,
+			      struct util_fabric, fabric_fid);
+#endif
+
+	buf->op_context = gnix_cq_err->op_context;
+	buf->flags = gnix_cq_err->flags;
+	buf->len = gnix_cq_err->len;
+	buf->buf = gnix_cq_err->buf;
+	buf->data = gnix_cq_err->data;
+	buf->tag = gnix_cq_err->tag;
+	buf->olen = gnix_cq_err->olen;
+	buf->err = gnix_cq_err->err;
+	buf->prov_errno = gnix_cq_err->prov_errno;
+
+	if (gnix_cq_err->err_data != NULL) {
+#if 0
+		if (buf->err_data_size != 0 &&
+		    FI_VERSION_GE(fabric->api_version, FI_VERSION(1, 5))) {
```

Comment:
As per the above comment re `struct util_fabric`, remove `fabric->api_version`.  You can just put something that doesn't compile in here like `GET_API_VERSION`.